From 232519b102571933f034cee72979ecf834af4b46 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 6 Sep 2015 14:14:03 +0200
Subject: [PATCH 2/9] portage: do not redefine builtin license

---
 backends/portage/portageBackend.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/backends/portage/portageBackend.py b/backends/portage/portageBackend.py
index d4726b6..f40ca92 100755
--- a/backends/portage/portageBackend.py
+++ b/backends/portage/portageBackend.py
@@ -334,14 +334,14 @@ class PackageKitPortageMixin(object):
     def _get_real_license_str(self, cpv, metadata):
         # use conditionals info (w/ USE) in LICENSE and remove ||
         ebuild_settings = self._get_ebuild_settings(cpv, metadata)
-        license = set(portage.flatten(
+        license_ = set(portage.flatten(
             portage.dep.use_reduce(
                 portage.dep.paren_reduce(metadata["LICENSE"]),
                 uselist=ebuild_settings.get("USE", "").split()
             )
         ))
-        license.discard('||')
-        return ' '.join(license)
+        license_.discard('||')
+        return ' '.join(license_)
 
     def _signal_config_update(self):
         result = list(portage.util.find_updated_config_files(
-- 
2.6.3

