From d581bea1699a8d9982e91a5de2b8dd315b6c2c54 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 3 May 2015 13:05:27 +0200
Subject: [PATCH 06/12] Simplify _get_cpv_slotted with collections module

---
 backends/portage/portageBackend.py | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/backends/portage/portageBackend.py b/backends/portage/portageBackend.py
index ddeca0d..773256a 100755
--- a/backends/portage/portageBackend.py
+++ b/backends/portage/portageBackend.py
@@ -26,6 +26,7 @@ import re
 import signal
 import sys
 import traceback
+from collections import defaultdict
 from itertools import izip
 
 # layman imports (>=2)
@@ -540,14 +541,11 @@ class PackageKitPortageMixin(object):
         return size
 
     def _get_cpv_slotted(self, cpv_list):
-        cpv_dict = {}
+        cpv_dict = defaultdict(list)
 
         for cpv in cpv_list:
             slot = self._get_metadata(cpv, ["SLOT"])[0]
-            if slot not in cpv_dict:
-                cpv_dict[slot] = [cpv]
-            else:
-                cpv_dict[slot].append(cpv)
+            cpv_dict[slot].append(cpv)
 
         return cpv_dict
 
-- 
2.3.6

