From 3c046f4586605f51a55c82f81861adcf5e408488 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 6 Sep 2015 16:10:32 +0200
Subject: [PATCH 8/9] python: set steps on PackagekitProgress instanciation

---
 lib/python/packagekit/progress.py | 18 ++++++++----------
 1 file changed, 8 insertions(+), 10 deletions(-)

diff --git a/lib/python/packagekit/progress.py b/lib/python/packagekit/progress.py
index d0c0c26..b5c7e52 100644
--- a/lib/python/packagekit/progress.py
+++ b/lib/python/packagekit/progress.py
@@ -34,24 +34,22 @@ class PackagekitProgress(Iterable):
     from packagekit import PackagekitProgress
 
     steps = [10, 30, 50, 70] # Milestones in %
-    progress = PackagekitProgress()
-    progress.set_steps(steps)
-    for milestone in range(len(steps)):
+    progress = PackagekitProgress(steps)
+    for milestone in progress:
         # do the action is this step
         for i in range(100):
             # do some action
-            print "progress : %s " % progress.percent
-        progress.step() # step to next milestone
-
+            print "progress : %s " % milestone
     '''
 
     #TODO: Add support for elapsed/remaining time
 
-    def __init__(self):
+    def __init__(self, steps=None):
         super(PackagekitProgress, self).__init__()
-        self.percent = 0
-        self.steps = []
-        self.current_step = 0
+        if not steps:
+            self.reset()
+        else:
+            self.set_steps(steps)
 
     def set_steps(self, steps):
         '''
-- 
2.6.3

