From 775b16156373b08a7bdd50c446d740051db93a8d Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 3 May 2015 14:27:08 +0200
Subject: [PATCH 08/12] Remove useless instruction

---
 backends/portage/portageBackend.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/backends/portage/portageBackend.py b/backends/portage/portageBackend.py
index fc2d45a..4ca4fb5 100755
--- a/backends/portage/portageBackend.py
+++ b/backends/portage/portageBackend.py
@@ -519,7 +519,6 @@ class PackageKitPortageMixin(object):
             size = self._get_metadata(cpv, ["SIZE"])[0]
             size = int(size) if size else 0
         else:
-            self
             metadata = self._get_metadata(cpv, ["IUSE", "SLOT"], in_dict=True)
 
             package = _emerge.Package.Package(
-- 
2.3.6

