From c08ca43ca45d55471f014dee93592b5448950aea Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 3 May 2015 17:54:59 +0200
Subject: [PATCH 12/12] Retrieve EAPI and KEYWORDS before calling setcpv method

---
 backends/portage/portageBackend.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/backends/portage/portageBackend.py b/backends/portage/portageBackend.py
index 9ce2b79..79b8449 100755
--- a/backends/portage/portageBackend.py
+++ b/backends/portage/portageBackend.py
@@ -938,7 +938,8 @@ class PackageKitPortageBackend(PackageKitPortageMixin, PackageKitBaseBackend):
                 continue
 
             metadata = self._get_metadata(
-                cpv, ["DESCRIPTION", "HOMEPAGE", "IUSE", "LICENSE", "SLOT"],
+                cpv, ["DESCRIPTION", "HOMEPAGE", "IUSE", "LICENSE", "SLOT",
+                      "EAPI", "KEYWORDS"],
                 in_dict=True
             )
 
@@ -1599,7 +1600,8 @@ class PackageKitPortageBackend(PackageKitPortageMixin, PackageKitBaseBackend):
                 match = True
                 metadata = self._get_metadata(
                     cpv, ["DESCRIPTION", "HOMEPAGE", "IUSE", "LICENSE",
-                          "repository", "SLOT"], in_dict=True
+                          "repository", "SLOT", "EAPI", "KEYWORDS"],
+                    in_dict=True
                 )
                 # update LICENSE to correspond to system settings
                 metadata["LICENSE"] = self._get_real_license_str(cpv, metadata)
-- 
2.3.6

