From 5862bb70240b1f311e0ace880e3fcf0af1c8c5f8 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 7 Sep 2016 01:31:15 +0200
Subject: [PATCH] Fix link failure with missing e_source_get_displayname symbol

---
 src/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 428226d..1032d29 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -98,8 +98,8 @@ gnome_todo_LDFLAGS = \
 	$(GNOME_TODO_WARN_LDFLAGS)
 
 gnome_todo_LDADD = \
-	$(GNOME_TODO_LIBS) \
-	$(top_builddir)/plugins/eds/libeds.a
+	$(top_builddir)/plugins/eds/libeds.a \
+	$(GNOME_TODO_LIBS)
 
 gnome_todo_DEPENDENCIES = \
 	$(top_builddir)/plugins/eds/libeds.a
-- 
2.9.3

